hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -1002,34 +1005,34 @@ update env ledger st ev = case (st, ev) of
   (Open OpenState{coordinatedHeadState = CoordinatedHeadState{confirmedSnapshot}, headId}, ClientEvent GetUTxO) ->
     -- TODO: Is it really intuitive that we respond from the confirmed ledger if
     -- transactions are validated against the seen ledger?
-    OnlyEffects [ClientEffect . GetUTxOResponse headId $ getField @"utxo" $ getSnapshot confirmedSnapshot]
+    Effects [ClientEffect . GetUTxOResponse headId $ getField @"utxo" $ getSnapshot confirmedSnapshot]
   -- Closed
   (Closed closedState, OnChainEvent Observation{observedTx = OnContestTx{snapshotNumber}}) ->
     onClosedChainContestTx closedState snapshotNumber
   (Closed cst@ClosedState{contestationDeadline, readyToFanoutSent, headId}, OnChainEvent Tick{chainTime})
     | chainTime > contestationDeadline && not readyToFanoutSent ->
         NewState
           (Closed cst{readyToFanoutSent = True})
-          [ClientEffect $ ReadyToFanout headId]
+          `Combined` Effects [ClientEffect $ ReadyToFanout headId]
   (Closed closedState, ClientEvent Fanout) ->
     onClosedClientFanout closedState
   (Closed closedState, OnChainEvent Observation{observedTx = OnFanoutTx{}, newChainState}) ->
     onClosedChainFanoutTx closedState newChainState
   -- General
   (currentState, OnChainEvent Rollback{rolledBackChainState}) ->
-    NewState (setChainState rolledBackChainState currentState) []
+    NewState (setChainState rolledBackChainState currentState)
   (Open ost@OpenState{}, OnChainEvent Tick{chainSlot}) ->
-    NewState (Open ost{currentSlot = chainSlot}) []
+    NewState (Open ost{currentSlot = chainSlot})
   (_, OnChainEvent Tick{}) ->
-    OnlyEffects []
+    Effects []
```

Comment:
Returning an empty list of effects is suspicious. Maybe we should introduce a `NoOutcome` notion